dask/dataframe/utils.py

```diff
@@ -459,7 +459,7 @@ def _check_dask(dsk, check_names=True, check_dtypes=True, result=None, scheduler
             graph.validate()
         if result is None:
             result = dsk.compute(scheduler=scheduler)
-        if isinstance(dsk, dd.Index):
+        if is_index_like(dsk._meta):
```

Comment:
Looks like `dask_cudf.Index` does **not** satisfy `is_index_like` when the underlying meta is a MultiIndex, because a `cudf.MultiIndex` does not have a `dtype` attribute (while `pd.MultiIndex` does) :/

I'm sure this can changed in cudf, but we could also do something like `if isinstance(dsk, dd.Index) or is_index_like(dsk._meta)`.